Version 8.0.0 and above incompatible with buildDashboard and projectReport

I am using Ktlint-gradle on a [big project](https://github.com/AlchemistSimulator/Alchemist/) of which I'm leading development.
I'm encountering issues when trying to update to 8.2.0. With newer ktlint I get:
```
$ ./gradlew projectReport --stacktrace

> Task :buildSrc:compileKotlin
The `kotlin-dsl` plugin applied to project ':buildSrc' enables experimental Kotlin compiler features. For more information see https://docs.gradle.org/5.6.2/userguide/kotlin_dsl.html#sec:kotlin-dsl_plugin

> Task :buildSrc:compileJava NO-SOURCE
> Task :buildSrc:compileGroovy NO-SOURCE
> Task :buildSrc:pluginDescriptors UP-TO-DATE
> Task :buildSrc:processResources NO-SOURCE
> Task :buildSrc:classes UP-TO-DATE
> Task :buildSrc:inspectClassesForKotlinIC UP-TO-DATE
> Task :buildSrc:jar UP-TO-DATE
> Task :buildSrc:assemble UP-TO-DATE
> Task :buildSrc:compileTestKotlin NO-SOURCE
> Task :buildSrc:pluginUnderTestMetadata UP-TO-DATE
> Task :buildSrc:compileTestJava NO-SOURCE
> Task :buildSrc:compileTestGroovy NO-SOURCE
> Task :buildSrc:processTestResources NO-SOURCE
> Task :buildSrc:testClasses UP-TO-DATE
> Task :buildSrc:test NO-SOURCE
> Task :buildSrc:validateTaskProperties UP-TO-DATE
> Task :buildSrc:check UP-TO-DATE
> Task :buildSrc:build UP-TO-DATE

> Configure project :
Website https://alchemistsimulator.github.io/latest/ is at version 9.0.0-dev1e+f56d2857. Orchid deployment enabled.

> Task :dependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/build/reports/project/dependencies.txt

> Task :htmlDependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/build/reports/project/dependencies/index.html

> Task :alchemist-cognitive-agents:htmlDependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-cognitive-agents/build/reports/project/dependencies/index.html

> Task :alchemist-cognitive-agents:buildDashboard UP-TO-DATE

> Task :alchemist-engine:htmlDependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-engine/build/reports/project/dependencies/index.html

> Task :alchemist-engine:buildDashboard UP-TO-DATE

> Task :alchemist-grid:htmlDependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-grid/build/reports/project/dependencies/index.html

> Task :alchemist-grid:buildDashboard UP-TO-DATE

> Task :alchemist-implementationbase:htmlDependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-implementationbase/build/reports/project/dependencies/index.html

> Task :alchemist-implementationbase:buildDashboard UP-TO-DATE

> Task :alchemist-incarnation-biochemistry:htmlDependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-incarnation-biochemistry/build/reports/project/dependencies/index.html

> Task :alchemist-incarnation-biochemistry:buildDashboard UP-TO-DATE

> Task :alchemist-incarnation-protelis:htmlDependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-incarnation-protelis/build/reports/project/dependencies/index.html

> Task :alchemist-incarnation-protelis:buildDashboard UP-TO-DATE

> Task :alchemist-incarnation-sapere:htmlDependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-incarnation-sapere/build/reports/project/dependencies/index.html

> Task :alchemist-incarnation-sapere:buildDashboard UP-TO-DATE

> Task :alchemist-incarnation-scafi:htmlDependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-incarnation-scafi/build/reports/project/dependencies/index.html

> Task :alchemist-incarnation-scafi:buildDashboard UP-TO-DATE

> Task :alchemist-influence-sphere:htmlDependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-influence-sphere/build/reports/project/dependencies/index.html

> Task :alchemist-influence-sphere:buildDashboard UP-TO-DATE

> Task :alchemist-interfaces:htmlDependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-interfaces/build/reports/project/dependencies/index.html

> Task :alchemist-interfaces:buildDashboard UP-TO-DATE

> Task :alchemist-loading:htmlDependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-loading/build/reports/project/dependencies/index.html

> Task :alchemist-loading:buildDashboard UP-TO-DATE

> Task :alchemist-maps:htmlDependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-maps/build/reports/project/dependencies/index.html

> Task :alchemist-maps:buildDashboard UP-TO-DATE

> Task :alchemist-projectview:htmlDependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-projectview/build/reports/project/dependencies/index.html

> Task :alchemist-projectview:buildDashboard UP-TO-DATE

> Task :alchemist-runner:htmlDependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-runner/build/reports/project/dependencies/index.html

> Task :alchemist-runner:buildDashboard UP-TO-DATE

> Task :alchemist-sapere-mathexp:htmlDependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-sapere-mathexp/build/reports/project/dependencies/index.html

> Task :alchemist-sapere-mathexp:buildDashboard UP-TO-DATE

> Task :alchemist-smartcam:htmlDependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-smartcam/build/reports/project/dependencies/index.html

> Task :alchemist-smartcam:buildDashboard UP-TO-DATE

> Task :alchemist-swingui:htmlDependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-swingui/build/reports/project/dependencies/index.html

> Task :alchemist-swingui:buildDashboard UP-TO-DATE

> Task :alchemist-time:htmlDependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-time/build/reports/project/dependencies/index.html

> Task :alchemist-time:buildDashboard UP-TO-DATE
> Task :buildDashboard FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':buildDashboard'.
> Could not create task ':ktlintMainSourceSetCheck'.
   > Cannot change dependencies of configuration ':ktlint' after it has been resolved.

* Try:
Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':buildDashboard'.
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:38)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:416)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:406)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:165)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:250)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:158)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:102)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.call(DelegatingBuildOperationExecutor.java:36)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
        at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:43)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:355)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:343)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:336)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:322)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker$1.execute(DefaultPlanExecutor.java:134)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker$1.execute(DefaultPlanExecutor.java:129)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:202)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.executeNextNode(DefaultPlanExecutor.java:193)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:129)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
Caused by: org.gradle.api.internal.tasks.DefaultTaskContainer$TaskCreationException: Could not create task ':ktlintMainSourceSetCheck'.
        at org.gradle.api.internal.tasks.DefaultTaskContainer.taskCreationException(DefaultTaskContainer.java:733)
        at org.gradle.api.internal.tasks.DefaultTaskContainer.access$600(DefaultTaskContainer.java:78)
        at org.gradle.api.internal.tasks.DefaultTaskContainer$TaskCreatingProvider.domainObjectCreationException(DefaultTaskContainer.java:725)
        at org.gradle.api.internal.DefaultNamedDomainObjectCollection$AbstractDomainObjectCreatingProvider.tryCreate(DefaultNamedDomainObjectCollection.java:946)
        at org.gradle.api.internal.tasks.DefaultTaskContainer$TaskCreatingProvider.access$1401(DefaultTaskContainer.java:671)
        at org.gradle.api.internal.tasks.DefaultTaskContainer$TaskCreatingProvider$1.run(DefaultTaskContainer.java:698)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:402)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:394)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:165)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:250)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:158)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:92)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
        at org.gradle.api.internal.tasks.DefaultTaskContainer$TaskCreatingProvider.tryCreate(DefaultTaskContainer.java:694)
        at org.gradle.api.internal.DefaultNamedDomainObjectCollection$AbstractDomainObjectCreatingProvider.getOrNull(DefaultNamedDomainObjectCollection.java:927)
        at org.gradle.api.internal.provider.AbstractReadOnlyProvider.get(AbstractReadOnlyProvider.java:27)
        at org.gradle.api.internal.DefaultNamedDomainObjectCollection$AbstractDomainObjectCreatingProvider.get(DefaultNamedDomainObjectCollection.java:913)
        at org.gradle.api.internal.provider.Collectors$ElementFromProvider.collectInto(Collectors.java:123)
        at org.gradle.api.internal.provider.Collectors$TypedCollector.collectInto(Collectors.java:371)
        at org.gradle.api.internal.collections.DefaultPendingSource.realize(DefaultPendingSource.java:58)
        at org.gradle.api.internal.collections.DefaultPendingSource.realizePending(DefaultPendingSource.java:49)
        at org.gradle.api.internal.collections.SortedSetElementSource.realizePending(SortedSetElementSource.java:111)
        at org.gradle.api.internal.DefaultDomainObjectCollection.addEagerAction(DefaultDomainObjectCollection.java:226)
        at org.gradle.api.internal.DefaultDomainObjectCollection.all(DefaultDomainObjectCollection.java:144)
        at org.gradle.api.reporting.GenerateBuildDashboard$2.execute(GenerateBuildDashboard.java:100)
        at org.gradle.api.reporting.GenerateBuildDashboard$2.execute(GenerateBuildDashboard.java:97)
        at org.gradle.api.internal.DefaultMutationGuard$2.execute(DefaultMutationGuard.java:42)
        at org.gradle.internal.Actions.with(Actions.java:251)
        at org.gradle.api.internal.project.BuildOperationCrossProjectConfigurator$2$1.run(BuildOperationCrossProjectConfigurator.java:79)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:402)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:394)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:165)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:250)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:158)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:92)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
        at org.gradle.api.internal.project.BuildOperationCrossProjectConfigurator$2.run(BuildOperationCrossProjectConfigurator.java:76)
        at org.gradle.internal.Factories$1.create(Factories.java:26)
        at org.gradle.api.internal.project.DefaultProjectStateRegistry$ProjectStateImpl.withMutableState(DefaultProjectStateRegistry.java:201)
        at org.gradle.api.internal.project.DefaultProjectStateRegistry$ProjectStateImpl.withMutableState(DefaultProjectStateRegistry.java:187)
        at org.gradle.api.internal.project.BuildOperationCrossProjectConfigurator.runProjectConfigureAction(BuildOperationCrossProjectConfigurator.java:73)
        at org.gradle.api.internal.project.BuildOperationCrossProjectConfigurator.access$400(BuildOperationCrossProjectConfigurator.java:32)
        at org.gradle.api.internal.project.BuildOperationCrossProjectConfigurator$1.doRunProjectConfigure(BuildOperationCrossProjectConfigurator.java:67)
        at org.gradle.api.internal.project.BuildOperationCrossProjectConfigurator$BlockConfigureBuildOperation.run(BuildOperationCrossProjectConfigurator.java:121)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:402)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:394)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:165)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:250)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:158)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:92)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
        at org.gradle.api.internal.project.BuildOperationCrossProjectConfigurator.runBlockConfigureAction(BuildOperationCrossProjectConfigurator.java:64)
        at org.gradle.api.internal.project.BuildOperationCrossProjectConfigurator.allprojects(BuildOperationCrossProjectConfigurator.java:54)
        at org.gradle.api.internal.project.DefaultProject.allprojects(DefaultProject.java:659)
        at org.gradle.api.reporting.GenerateBuildDashboard.getAggregatedTasks(GenerateBuildDashboard.java:97)
        at org.gradle.api.reporting.GenerateBuildDashboard.getEnabledInputReports(GenerateBuildDashboard.java:83)
        at org.gradle.api.reporting.GenerateBuildDashboard.getInputReports(GenerateBuildDashboard.java:71)
        at org.gradle.api.reporting.GenerateBuildDashboard_Decorated.getInputReports(Unknown Source)
        at org.gradle.api.internal.tasks.properties.bean.AbstractNestedRuntimeBeanNode$BeanPropertyValue$1$1.create(AbstractNestedRuntimeBeanNode.java:80)
        at org.gradle.util.SingleMessageLogger.whileDisabled(SingleMessageLogger.java:467)
        at org.gradle.api.internal.tasks.properties.bean.AbstractNestedRuntimeBeanNode$BeanPropertyValue$1.get(AbstractNestedRuntimeBeanNode.java:76)
        at com.google.common.base.Suppliers$NonSerializableMemoizingSupplier.get(Suppliers.java:167)
        at org.gradle.api.internal.tasks.properties.bean.AbstractNestedRuntimeBeanNode$BeanPropertyValue.call(AbstractNestedRuntimeBeanNode.java:148)
        at org.gradle.api.internal.tasks.properties.AbstractValidatingProperty.validate(AbstractValidatingProperty.java:37)
        at org.gradle.api.internal.tasks.properties.DefaultTaskProperties.validate(DefaultTaskProperties.java:193)
        at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:108)
        at org.gradle.api.internal.tasks.execution.ResolveBeforeExecutionOutputsTaskExecuter.execute(ResolveBeforeExecutionOutputsTaskExecuter.java:67)
        at org.gradle.api.internal.tasks.execution.StartSnapshotTaskInputsBuildOperationTaskExecuter.execute(StartSnapshotTaskInputsBuildOperationTaskExecuter.java:62)
        at org.gradle.api.internal.tasks.execution.ResolveAfterPreviousExecutionStateTaskExecuter.execute(ResolveAfterPreviousExecutionStateTaskExecuter.java:46)
        at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:94)
        at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
        at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:95)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:56)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
        ... 24 more
Caused by: org.gradle.api.InvalidUserDataException: Cannot change dependencies of configuration ':ktlint' after it has been resolved.
        at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration.preventIllegalMutation(DefaultConfiguration.java:1127)
        at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration.validateMutation(DefaultConfiguration.java:1098)
        at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration$3.execute(DefaultConfiguration.java:299)
        at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration$3.execute(DefaultConfiguration.java:296)
        at org.gradle.internal.ImmutableActionSet$SingletonSet.execute(ImmutableActionSet.java:225)
        at org.gradle.api.internal.DefaultDomainObjectSet.assertMutableCollectionContents(DefaultDomainObjectSet.java:65)
        at org.gradle.api.internal.DefaultDomainObjectCollection.addLater(DefaultDomainObjectCollection.java:284)
        at org.gradle.api.internal.DelegatingDomainObjectSet.addLater(DelegatingDomainObjectSet.java:100)
        at org.jlleitschuh.gradle.ktlint.ConfigurationsKt.createKtlintConfiguration(Configurations.kt:26)
        at org.jlleitschuh.gradle.ktlint.KtlintPlugin$PluginHolder$ktlintConfiguration$2.invoke(KtlintPlugin.kt:385)
        at org.jlleitschuh.gradle.ktlint.KtlintPlugin$PluginHolder$ktlintConfiguration$2.invoke(KtlintPlugin.kt:366)
        at kotlin.UnsafeLazyImpl.getValue(Lazy.kt:81)
        at org.jlleitschuh.gradle.ktlint.KtlintPlugin$PluginHolder.getKtlintConfiguration(KtlintPlugin.kt)
        at org.jlleitschuh.gradle.ktlint.KtlintPlugin.configurePluginTask(KtlintPlugin.kt:279)
        at org.jlleitschuh.gradle.ktlint.KtlintPlugin.access$configurePluginTask(KtlintPlugin.kt:27)
        at org.jlleitschuh.gradle.ktlint.KtlintPlugin$createCheckTask$1.invoke(KtlintPlugin.kt:269)
        at org.jlleitschuh.gradle.ktlint.KtlintPlugin$createCheckTask$1.invoke(KtlintPlugin.kt:27)
        at org.jlleitschuh.gradle.ktlint.PluginUtilKt$sam$i$org_gradle_api_Action$0.execute(PluginUtil.kt)
        at org.gradle.api.internal.DefaultMutationGuard$2.execute(DefaultMutationGuard.java:42)
        at org.gradle.api.internal.DefaultMutationGuard$2.execute(DefaultMutationGuard.java:42)
        at org.gradle.api.internal.DefaultCollectionCallbackActionDecorator$BuildOperationEmittingAction$1$1.run(DefaultCollectionCallbackActionDecorator.java:100)
        at org.gradle.configuration.internal.DefaultUserCodeApplicationContext.reapply(DefaultUserCodeApplicationContext.java:60)
        at org.gradle.api.internal.DefaultCollectionCallbackActionDecorator$BuildOperationEmittingAction$1.run(DefaultCollectionCallbackActionDecorator.java:97)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:402)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:394)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:165)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:250)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:158)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:92)
        at org.gradle.api.internal.DefaultCollectionCallbackActionDecorator$BuildOperationEmittingAction.execute(DefaultCollectionCallbackActionDecorator.java:94)
        at org.gradle.internal.ImmutableActionSet$SetWithManyActions.execute(ImmutableActionSet.java:329)
        at org.gradle.api.internal.DefaultDomainObjectCollection.doAdd(DefaultDomainObjectCollection.java:264)
        at org.gradle.api.internal.DefaultNamedDomainObjectCollection.doAdd(DefaultNamedDomainObjectCollection.java:112)
        at org.gradle.api.internal.DefaultDomainObjectCollection.add(DefaultDomainObjectCollection.java:258)
        at org.gradle.api.internal.DefaultNamedDomainObjectCollection$AbstractDomainObjectCreatingProvider.tryCreate(DefaultNamedDomainObjectCollection.java:942)
        ... 96 more


* Get more help at https://help.gradle.org

BUILD FAILED in 5s
39 actionable tasks: 21 executed, 18 up-to-date
```
But the same build succeeds with older versions, e.g. with 7.4.0:
```
$ ./gradlew projectReport --stacktrace

> Task :buildSrc:compileKotlin
The `kotlin-dsl` plugin applied to project ':buildSrc' enables experimental Kotlin compiler features. For more information see https://docs.gradle.org/5.6.2/userguide/kotlin_dsl.html#sec:kotlin-dsl_plugin
The `kotlin-dsl` plugin applied to project ':buildSrc' enables experimental Kotlin compiler features. For more information see https://docs.gradle.org/5.6.2/userguide/kotlin_dsl.html#sec:kotlin-dsl_plugin

> Task :buildSrc:compileJava NO-SOURCE
> Task :buildSrc:compileGroovy NO-SOURCE
> Task :buildSrc:pluginDescriptors UP-TO-DATE
> Task :buildSrc:processResources NO-SOURCE
> Task :buildSrc:classes UP-TO-DATE
> Task :buildSrc:inspectClassesForKotlinIC

> Task :buildSrc:jar
:jar: No valid plugin descriptors were found in META-INF/gradle-plugins

> Task :buildSrc:assemble
> Task :buildSrc:compileTestKotlin NO-SOURCE
> Task :buildSrc:pluginUnderTestMetadata
> Task :buildSrc:compileTestJava NO-SOURCE
> Task :buildSrc:compileTestGroovy NO-SOURCE
> Task :buildSrc:processTestResources NO-SOURCE
> Task :buildSrc:testClasses UP-TO-DATE
> Task :buildSrc:test NO-SOURCE
> Task :buildSrc:validateTaskProperties
> Task :buildSrc:check
> Task :buildSrc:build

> Configure project :
Website https://alchemistsimulator.github.io/latest/ is at version 9.0.0-dev1e+f56d2857. Orchid deployment enabled.

> Task :dependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/build/reports/project/dependencies.txt

> Task :htmlDependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/build/reports/project/dependencies/index.html

> Task :alchemist-cognitive-agents:htmlDependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-cognitive-agents/build/reports/project/dependencies/index.html

> Task :alchemist-cognitive-agents:buildDashboard UP-TO-DATE

> Task :alchemist-engine:htmlDependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-engine/build/reports/project/dependencies/index.html

> Task :alchemist-engine:buildDashboard UP-TO-DATE

> Task :alchemist-grid:htmlDependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-grid/build/reports/project/dependencies/index.html

> Task :alchemist-grid:buildDashboard UP-TO-DATE

> Task :alchemist-implementationbase:htmlDependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-implementationbase/build/reports/project/dependencies/index.html

> Task :alchemist-implementationbase:buildDashboard UP-TO-DATE

> Task :alchemist-incarnation-biochemistry:htmlDependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-incarnation-biochemistry/build/reports/project/dependencies/index.html

> Task :alchemist-incarnation-biochemistry:buildDashboard UP-TO-DATE

> Task :alchemist-incarnation-protelis:htmlDependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-incarnation-protelis/build/reports/project/dependencies/index.html

> Task :alchemist-incarnation-protelis:buildDashboard UP-TO-DATE

> Task :alchemist-incarnation-sapere:htmlDependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-incarnation-sapere/build/reports/project/dependencies/index.html

> Task :alchemist-incarnation-sapere:buildDashboard UP-TO-DATE

> Task :alchemist-incarnation-scafi:htmlDependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-incarnation-scafi/build/reports/project/dependencies/index.html

> Task :alchemist-incarnation-scafi:buildDashboard UP-TO-DATE

> Task :alchemist-influence-sphere:htmlDependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-influence-sphere/build/reports/project/dependencies/index.html

> Task :alchemist-influence-sphere:buildDashboard UP-TO-DATE

> Task :alchemist-interfaces:htmlDependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-interfaces/build/reports/project/dependencies/index.html

> Task :alchemist-interfaces:buildDashboard UP-TO-DATE

> Task :alchemist-loading:htmlDependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-loading/build/reports/project/dependencies/index.html

> Task :alchemist-loading:buildDashboard UP-TO-DATE

> Task :alchemist-maps:htmlDependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-maps/build/reports/project/dependencies/index.html

> Task :alchemist-maps:buildDashboard UP-TO-DATE

> Task :alchemist-projectview:htmlDependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-projectview/build/reports/project/dependencies/index.html

> Task :alchemist-projectview:buildDashboard UP-TO-DATE

> Task :alchemist-runner:htmlDependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-runner/build/reports/project/dependencies/index.html

> Task :alchemist-runner:buildDashboard UP-TO-DATE

> Task :alchemist-sapere-mathexp:htmlDependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-sapere-mathexp/build/reports/project/dependencies/index.html

> Task :alchemist-sapere-mathexp:buildDashboard UP-TO-DATE

> Task :alchemist-smartcam:htmlDependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-smartcam/build/reports/project/dependencies/index.html

> Task :alchemist-smartcam:buildDashboard UP-TO-DATE

> Task :alchemist-swingui:htmlDependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-swingui/build/reports/project/dependencies/index.html

> Task :alchemist-swingui:buildDashboard UP-TO-DATE

> Task :alchemist-time:htmlDependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-time/build/reports/project/dependencies/index.html

> Task :alchemist-time:buildDashboard UP-TO-DATE
> Task :buildDashboard UP-TO-DATE

> Task :propertyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/build/reports/project/properties.txt

> Task :taskReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/build/reports/project/tasks.txt

> Task :projectReport

> Task :alchemist-cognitive-agents:dependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-cognitive-agents/build/reports/project/dependencies.txt

> Task :alchemist-cognitive-agents:propertyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-cognitive-agents/build/reports/project/properties.txt

> Task :alchemist-cognitive-agents:taskReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-cognitive-agents/build/reports/project/tasks.txt

> Task :alchemist-cognitive-agents:projectReport

> Task :alchemist-engine:dependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-engine/build/reports/project/dependencies.txt

> Task :alchemist-engine:propertyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-engine/build/reports/project/properties.txt

> Task :alchemist-engine:taskReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-engine/build/reports/project/tasks.txt

> Task :alchemist-engine:projectReport

> Task :alchemist-grid:dependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-grid/build/reports/project/dependencies.txt

> Task :alchemist-grid:propertyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-grid/build/reports/project/properties.txt

> Task :alchemist-grid:taskReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-grid/build/reports/project/tasks.txt

> Task :alchemist-grid:projectReport

> Task :alchemist-implementationbase:dependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-implementationbase/build/reports/project/dependencies.txt

> Task :alchemist-implementationbase:propertyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-implementationbase/build/reports/project/properties.txt

> Task :alchemist-implementationbase:taskReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-implementationbase/build/reports/project/tasks.txt

> Task :alchemist-implementationbase:projectReport

> Task :alchemist-incarnation-biochemistry:dependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-incarnation-biochemistry/build/reports/project/dependencies.txt

> Task :alchemist-incarnation-biochemistry:propertyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-incarnation-biochemistry/build/reports/project/properties.txt

> Task :alchemist-incarnation-biochemistry:taskReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-incarnation-biochemistry/build/reports/project/tasks.txt

> Task :alchemist-incarnation-biochemistry:projectReport

> Task :alchemist-incarnation-protelis:dependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-incarnation-protelis/build/reports/project/dependencies.txt

> Task :alchemist-incarnation-protelis:propertyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-incarnation-protelis/build/reports/project/properties.txt

> Task :alchemist-incarnation-protelis:taskReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-incarnation-protelis/build/reports/project/tasks.txt

> Task :alchemist-incarnation-protelis:projectReport

> Task :alchemist-incarnation-sapere:dependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-incarnation-sapere/build/reports/project/dependencies.txt

> Task :alchemist-incarnation-sapere:propertyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-incarnation-sapere/build/reports/project/properties.txt

> Task :alchemist-incarnation-sapere:taskReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-incarnation-sapere/build/reports/project/tasks.txt

> Task :alchemist-incarnation-sapere:projectReport

> Task :alchemist-incarnation-scafi:dependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-incarnation-scafi/build/reports/project/dependencies.txt

> Task :alchemist-incarnation-scafi:propertyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-incarnation-scafi/build/reports/project/properties.txt

> Task :alchemist-incarnation-scafi:taskReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-incarnation-scafi/build/reports/project/tasks.txt

> Task :alchemist-incarnation-scafi:projectReport

> Task :alchemist-influence-sphere:dependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-influence-sphere/build/reports/project/dependencies.txt

> Task :alchemist-influence-sphere:propertyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-influence-sphere/build/reports/project/properties.txt

> Task :alchemist-influence-sphere:taskReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-influence-sphere/build/reports/project/tasks.txt

> Task :alchemist-influence-sphere:projectReport

> Task :alchemist-interfaces:dependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-interfaces/build/reports/project/dependencies.txt

> Task :alchemist-interfaces:propertyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-interfaces/build/reports/project/properties.txt

> Task :alchemist-interfaces:taskReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-interfaces/build/reports/project/tasks.txt

> Task :alchemist-interfaces:projectReport

> Task :alchemist-loading:dependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-loading/build/reports/project/dependencies.txt

> Task :alchemist-loading:propertyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-loading/build/reports/project/properties.txt

> Task :alchemist-loading:taskReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-loading/build/reports/project/tasks.txt

> Task :alchemist-loading:projectReport

> Task :alchemist-maps:dependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-maps/build/reports/project/dependencies.txt

> Task :alchemist-maps:propertyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-maps/build/reports/project/properties.txt

> Task :alchemist-maps:taskReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-maps/build/reports/project/tasks.txt

> Task :alchemist-maps:projectReport

> Task :alchemist-projectview:dependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-projectview/build/reports/project/dependencies.txt

> Task :alchemist-projectview:propertyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-projectview/build/reports/project/properties.txt

> Task :alchemist-projectview:taskReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-projectview/build/reports/project/tasks.txt

> Task :alchemist-projectview:projectReport

> Task :alchemist-runner:dependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-runner/build/reports/project/dependencies.txt

> Task :alchemist-runner:propertyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-runner/build/reports/project/properties.txt

> Task :alchemist-runner:taskReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-runner/build/reports/project/tasks.txt

> Task :alchemist-runner:projectReport

> Task :alchemist-sapere-mathexp:dependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-sapere-mathexp/build/reports/project/dependencies.txt

> Task :alchemist-sapere-mathexp:propertyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-sapere-mathexp/build/reports/project/properties.txt

> Task :alchemist-sapere-mathexp:taskReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-sapere-mathexp/build/reports/project/tasks.txt

> Task :alchemist-sapere-mathexp:projectReport

> Task :alchemist-smartcam:dependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-smartcam/build/reports/project/dependencies.txt

> Task :alchemist-smartcam:propertyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-smartcam/build/reports/project/properties.txt

> Task :alchemist-smartcam:taskReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-smartcam/build/reports/project/tasks.txt

> Task :alchemist-smartcam:projectReport

> Task :alchemist-swingui:dependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-swingui/build/reports/project/dependencies.txt

> Task :alchemist-swingui:propertyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-swingui/build/reports/project/properties.txt

> Task :alchemist-swingui:taskReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-swingui/build/reports/project/tasks.txt

> Task :alchemist-swingui:projectReport

> Task :alchemist-time:dependencyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-time/build/reports/project/dependencies.txt

> Task :alchemist-time:propertyReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-time/build/reports/project/properties.txt

> Task :alchemist-time:taskReport
See the report at: file:///home/danysk/Dropbox/Workspaces/Alchemist/alchemist/alchemist-time/build/reports/project/tasks.txt

> Task :alchemist-time:projectReport

BUILD SUCCESSFUL in 9s
95 actionable tasks: 76 executed, 19 up-to-date
```

Is there something I'm missing which is required for upgrading? The checker, if ran alone, works without any issue.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -13,7 +13,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/).
 ### Removed
   - ?
 ### Fixed
-  - ?
+  - Ktlint configuration could be tried to configured after been resolved (#283)
 
 ## [9.0.0] - 2019-09-30
 ### Added

--- a/plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/Configurations.kt
+++ b/plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/Configurations.kt
@@ -2,6 +2,7 @@ package org.jlleitschuh.gradle.ktlint
 
 import net.swiftzer.semver.SemVer
 import org.gradle.api.Project
+import org.gradle.api.artifacts.Configuration
 import org.gradle.api.artifacts.Dependency
 
 internal const val KTLINT_CONFIGURATION_NAME = "ktlint"
@@ -13,6 +14,9 @@ internal const val KTLINT_REPORTER_CONFIGURATION_DESCRIPTION = "All ktlint custo
 
 internal fun createKtlintConfiguration(target: Project, extension: KtlintExtension) =
     target.configurations.maybeCreate(KTLINT_CONFIGURATION_NAME).apply {
+        // Configurations in resolved state are not allowed to modify dependencies
+        if (state != Configuration.State.UNRESOLVED) return@apply
+
         description = KTLINT_CONFIGURATION_DESCRIPTION
         val dependencyProvider = target.provider<Dependency> {
             val ktlintVersion = extension.version.get()

